src/test/java/pl/touk/sputnik/processor/shellcheck/ShellcheckProcessorTest.java

```diff
@@ -0,0 +1,104 @@
+
+    private static final String CONFIGURATION_WITH_SHELLCHECK_ENABLED = "shellcheck/sputnik/noConfigurationFile.properties";
+    private static final String REVIEW_FILE_WITH_ONE_VIOLATION = "src/test/resources/shellcheck/testFiles/oneViolation.sh";
+    private static final String REVIEW_FILE_WITH_MULTIPLE_VIOLATIONS = "src/test/resources/shellcheck/testFiles/multipleViolations.sh";
+    private ShellcheckProcessor sut;
+
+    @Before
+    public void setUp() {
+        config = ConfigurationBuilder.initFromResource(CONFIGURATION_WITH_SHELLCHECK_ENABLED);
+        sut = new ShellcheckProcessor(config);
+    }
+
+    @Test
+    public void shouldReturnNoViolationsWhenThereIsNoFileToReview() {
+        ReviewResult reviewResult = sut.process(nonexistantReview());
+
+        assertThat(reviewResult).isNotNull();
+        assertThat(reviewResult.getViolations()).isEmpty();
+    }
+
+    @Test
+    public void shouldReturnOneViolationsForFile() {
+        Assume.assumeTrue(isShellcheckInstalled());
+        Review review = getReview(REVIEW_FILE_WITH_ONE_VIOLATION);
+
+        ReviewResult result = sut.process(review);
+
+        assertThat(result).isNotNull();
+        assertThat(result.getViolations())
+                .hasSize(1);
+        Violation violation = result.getViolations().get(0);
+        assertThat(violation.getFilenameOrJavaClassName()).contains(REVIEW_FILE_WITH_ONE_VIOLATION);
+        assertThat(violation.getLine()).isEqualTo(3);
+        assertThat(violation.getMessage()).isEqualTo("[Code:2154] Message: variable is referenced but not assigned.");
+        assertThat(violation.getSeverity()).isEqualTo(Severity.WARNING);
+    }
+
+    @Test
+    public void shouldReturnMultipleViolationsForFile() {
+        Assume.assumeTrue(isShellcheckInstalled());
+        Review review = getReview(REVIEW_FILE_WITH_MULTIPLE_VIOLATIONS);
+
+        ReviewResult result = sut.process(review);
+
+        assertThat(result).isNotNull();
+        assertThat(result.getViolations())
+                .hasSize(3);
+        Violation violation1 = result.getViolations().get(0);
+        assertThat(violation1.getFilenameOrJavaClassName()).contains(REVIEW_FILE_WITH_MULTIPLE_VIOLATIONS);
+        assertThat(violation1.getLine()).isEqualTo(3);
+        assertThat(violation1.getMessage()).isEqualTo("[Code:2154] Message: variable is referenced but not assigned.");
+        assertThat(violation1.getSeverity()).isEqualTo(Severity.WARNING);
+
+        Violation violation2 = result.getViolations().get(1);
+        assertThat(violation2.getFilenameOrJavaClassName()).contains(REVIEW_FILE_WITH_MULTIPLE_VIOLATIONS);
+        assertThat(violation2.getLine()).isEqualTo(5);
+        assertThat(violation2.getMessage()).isEqualTo("[Code:1037] Message: Braces are required for positionals over 9, e.g. ${10}.");
+        assertThat(violation2.getSeverity()).isEqualTo(Severity.ERROR);
+
+        Violation violation3= result.getViolations().get(2);
```

Comment:
\[Checkstyle\] ERROR: '=' is not preceded with whitespace\.

Reply:
+1